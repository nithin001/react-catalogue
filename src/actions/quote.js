import { put } from '../api/cart';
export const load = (cart) => {
  return (dispatch) => {
    dispatch({ type: 'LOAD_QUOTE_START' });
    return put(cart)
      .then((quote) => {
        dispatch({ type: 'POPULATE_QUOTE', payload: quote });
        dispatch({ type: 'LOAD_QUOTE_COMPLETE' });
      })
      .catch(() => {
        dispatch({ type: 'LOAD_ERROR' });
      });
  };
};