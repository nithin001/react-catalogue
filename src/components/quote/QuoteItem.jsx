import React from 'react';

export const QuoteItem = ({ order }) => (
  <div className={'quote-item'}>
    <span className={'quote-item__name'}>
      <span className={'quote-item__name quote-item__name--legend'}>Name:&nbsp;</span>
      <span className={'quote-item__name quote-item__name--value'}>{order.get('name')}</span>
    </span>
    <br/>
    <span className={'quote-item__price'}>
      <span className={'quote-item__name quote-item__name--legend'}>Price:&nbsp;</span>
      <span className={'quote-item__price quote-item__price--amount'}>{order.getIn(['price', 'amount'])}</span>
      &nbsp;
      <span className={'quote-item__price quote-item__price--currency'}>{order.getIn(['price', 'currency'])}</span>
    </span>
    <br/>
    <span className={'quote-item__total-quantity'}>
      <span className={'quote-item__quantity'}>
        <span className={'quote-item__quantity quote-item__quantity--legend'}>Quantity:&nbsp;</span>
        <span className={'quote-item__quantity quote-item__quantity--value'}>{order.get('quantity')}</span>
      </span>
      <span className={'quote-item__total'}>
        <span className={'quote-item__total quote-item__total--amount'}>{order.getIn(['lineTotal', 'amount'])}</span>
        &nbsp;
        <span className={'quote-item__total quote-item__total--currency'}>{order.getIn(
          ['lineTotal', 'currency'])}</span>
      </span>
    </span>
  </div>
);