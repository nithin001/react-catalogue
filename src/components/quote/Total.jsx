import React from 'react';

export const Total = ({ total }) => (
  <span className={'quote-total'}>
    <span className={'quote-total__legend'}>Total:&nbsp;</span>
    <span className={'quote-total__value'}>
      <span className={'quote-total__value quote-total__value--amount'}>{total.get('amount')}</span>
      &nbsp;
      <span className={'quote-total__value quote-total__value--currency'}>{total.get('currency')}</span>
    </span>
  </span>
);