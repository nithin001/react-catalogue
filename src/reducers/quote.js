import Immutable from 'immutable';
const quote = (state = Immutable.Map(), action) => {
  if (action.type === 'POPULATE_QUOTE') {
    return Immutable.fromJS(action.payload);
  }
  return state;
};
export default quote;