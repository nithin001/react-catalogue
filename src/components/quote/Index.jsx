import React from 'react';
import { connect } from 'react-redux';
import { Total } from './Total';
import { QuoteItem } from './QuoteItem';
export const Quote = ({ quote, loading }) => {
  if (loading) {
    return <span>Loading Cart</span>;
  }
  if (quote.get('lines') === undefined || quote.get('lines').size == 0) {
    return <span>Quotation is empty</span>;
  }
  const quoteItems = quote.get('lines').map(quoteItem => {
    return (<QuoteItem key={quoteItem.get('sku')} order={quoteItem}/>);
  });

  return (
    <div>
      <div className={'quote-items'}>
        {quoteItems}
      </div>
      <Total total={quote.get('total')}/>
    </div>
  );
};

export const mapStateToProps = (state) => {
  return { quote: state.quote };
};

export default connect(mapStateToProps, null)(Quote);
