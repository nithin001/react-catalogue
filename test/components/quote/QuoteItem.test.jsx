import React from 'react';
import renderer from 'react-test-renderer';
import quote from '../../../src/reducers/quote';
import { QuoteItem } from '../../../src/components/quote/QuoteItem';
var mockQuote = {
  'lines': [{
    'sku': '374847',
    'name': 'Nice Product',
    'price': {
      'amount': 20,
      'currency': 'EUR'
    },
    'quantity': 3,
    'lineTotal': {
      'amount': 60,
      'currency': 'EUR'
    }
  }],
  'total': {
    'amount': 60,
    'currency': 'EUR'
  }
};
const quoteState = quote(undefined, { type: 'POPULATE_QUOTE', payload: mockQuote });
describe('QuoteItem component', () => {
  it('should render the quote item with name, price, quantity and line total', () => {
    const props = {
      order: quoteState.getIn(['lines', 0])
    };
    const quoteItem = renderer.create(<QuoteItem
      {...props}
    />);
    expect(quoteItem).toMatchSnapshot();
  });
});