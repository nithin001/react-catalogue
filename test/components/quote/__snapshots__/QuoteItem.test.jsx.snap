// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuoteItem component should render the quote item with name, price, quantity and line total 1`] = `
<div
  className="quote-item"
>
  <span
    className="quote-item__name"
  >
    <span
      className="quote-item__name quote-item__name--legend"
    >
      Name: 
    </span>
    <span
      className="quote-item__name quote-item__name--value"
    >
      Nice Product
    </span>
  </span>
  <br />
  <span
    className="quote-item__price"
  >
    <span
      className="quote-item__name quote-item__name--legend"
    >
      Price: 
    </span>
    <span
      className="quote-item__price quote-item__price--amount"
    >
      20
    </span>
     
    <span
      className="quote-item__price quote-item__price--currency"
    >
      EUR
    </span>
  </span>
  <br />
  <span
    className="quote-item__total-quantity"
  >
    <span
      className="quote-item__quantity"
    >
      <span
        className="quote-item__quantity quote-item__quantity--legend"
      >
        Quantity: 
      </span>
      <span
        className="quote-item__quantity quote-item__quantity--value"
      >
        3
      </span>
    </span>
    <span
      className="quote-item__total"
    >
      <span
        className="quote-item__total quote-item__total--amount"
      >
        60
      </span>
       
      <span
        className="quote-item__total quote-item__total--currency"
      >
        EUR
      </span>
    </span>
  </span>
</div>
`;
